flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/EdgeManagerBuildUtil.java

```diff
@@ -58,6 +58,26 @@ static void connectVertexToResult(
         }
     }
 
+    /**
+     * Given parallelisms of two job vertices, calculate the max number of input or output edges of
+     * each execution in the first job vertex.
+     *
+     * @param thisParallelism parallelism of the first job vertex.
+     * @param targetParallelism parallelism of the second job vertex.
+     * @param distributionPattern the {@link DistributionPattern} of the connecting edge.
+     */
+    public static int getMaxNumEdgesToTarget(
```

Comment:
I find the name is a bit confusing although the doc is clear.
But I do not have any good idea yet.

Reply:
To be honest, I also think it's confusing, but didn't find a better one.

Reply:
Maybe `getMaxEdgesToTargetExecutionVertex(int targetParallelism, int sourceParallelism, DistributionPattern distributionPattern)`?
And the java docs can be "Given parallelisms of two job vertices, calculate the max number of edges connected to a target execution vertex from the source execution vertices. Note that edges is considered undirected here. It can be an edge connected from an upstream job vertex to a downstream job vertex, or in a reversed way."